elrond-wasm/src/types/interaction/contract_call.rs

```diff
@@ -46,9 +52,13 @@ where
     SA: SendApi + 'static,
 {
     let mut contract_call = ContractCall::<SA, R>::new(api, to, endpoint_name);
-    contract_call.payment_token = payment_token;
-    contract_call.payment_amount = payment_amount;
-    contract_call.payment_nonce = payment_nonce;
+    let payment = EsdtTokenPayment {
+        token_type: EsdtTokenType::Invalid,
```

Comment:
Might not be such a great idea to use `EsdtTokenType::Invalid` here, even though currently this field isn't really used. I think the best approach would be to add a static method in `EsdtTokenPayment`, something like `from_fields(token_id, nonce, amount) -> Self`, which automatically sets `token_type` accordingly and returns a new instance.